### PR TITLE
create route signup

### PR DESCRIPTION
**Pull request by yousrakhaleel**
create route signup , signup validation by joi , sign up controller, fix eslint errors
Relates #15

### CODE REVIEW

File: server/src/controllers/users/signUp.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  try {
+    signupValidation(req.body).then(async () => {
+      const { email, password, confirmPassword } = req.body;
+      const isEmailExist = await User.findOne({ where: { email } });
+      if (isEmailExist) {
+        res.status(400).json({ success: false, message: 'Email is already exist !' });
+      } else if (confirmPassword === password) {
+        bcrypt.hash(password, 10).then(async (encryptedPass) => {
+          // eslint-disable-next-line max-len
+          const user = await User.create({ ...req.body, password: encryptedPass, confirmPassword: encryptedPass });
```
Comment (aaamra) [inline server/src/controllers/users/signUp.ts:17]: you don't have to store confirmPassword in the database it's just for validation

**Review by MahmoudJD95 — DISMISSED** [review]

**Review by SalmaIssa96 — DISMISSED** [review]

File: server/src/database/config/connection.ts [hunk 2]
```diff
@@ -25,6 +27,10 @@ if (!connectionString) throw new Error('no db url, check .env file for valid key
 const sequelize = new Sequelize(connectionString, {
   dialect: 'postgres',
   dialectOptions: { ssl },
+  logging: false,
+  define: {
+    timestamps: false,
+  },
```
Comment (aaamra) [inline server/src/database/config/connection.ts:33]: why did you disable the `timestamps` ?
i think it's useful

File: server/src/database/models/Product.ts [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+    type: DataTypes.DECIMAL,
+    allowNull: false,
+  },
+  details: {
+    type: DataTypes.TEXT,
+  },
+//   quantity:{
+//     type: DataTypes.INTEGER,
+//     allowNull: false
+//   }
```
Comment (aaamra) [inline server/src/database/models/Product.ts:28]: delete this

**Review by aaamra — APPROVED** [review]

**Review by MahmoudJD95 — APPROVED** [review]

File: server/src/controllers/users/signUp.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+          jwtSign({ id: user.id, isAdmin: user.is_admin })
+            .then((token) => {
+              res.cookie('token', token)
+                .status(201)
+                .json({ success: true });
+            });
+        }).catch(() => {
+          next('Internal Server Error');
```
Comment (Ibrahim-Jarada) [inline server/src/controllers/users/signUp.ts:25]: Same as above you need to think of  try catch block

`try{

}catch(err){

}`

**Review by Ibrahim-Jarada — COMMENTED** [review]
Consider these  changes

**Pull request merged**